p4rt_client/p4rt_client.go

```diff
@@ -346,6 +348,51 @@ func (p *P4RTClientStream) GetPacket(minSeqNum uint64) (uint64, *P4RTPacketInfo,
 	return p.pktCounters.RxPktCntr, pktInfo, nil
 }
 
+func (p *P4RTClientStream) GetPacketsWithTimeout(minSeqNum uint64, timeout time.Duration) (uint64, []*P4RTPacketInfo, error) {
+
+	done := make(chan struct{})
+	defer close(done)
+	var failed atomic.Bool
+
+	// routine returns directly if 'done' else updates 'failed'
+	// and signals pktCond.
+	go func() {
+		select {
+		case <-done:
+			return
+		case <-time.After(timeout):
+			failed.Store(true)
+			p.pktCond.Signal()
+		}
+	}()
+
+	p.pkt_mu.Lock()
+	defer p.pkt_mu.Unlock()
+	for p.pktCounters.RxPktCntr < minSeqNum {
+		// if timeout then loop breaks here.
+		if failed.Load() {
+			break
+		}
+		if glog.V(2) {
+			glog.Infof("'%s' Waiting on Packet (%d/%d)\n",
+				p, p.pktCounters.RxPktCntr, minSeqNum)
+		}
+		if p.ShouldStop() {
+			return 0, nil, io.EOF
+		}
+		p.pktCond.Wait()
+	}
+
+	var pkts []*P4RTPacketInfo
+
+	for i := 0; i < len(p.pktQ); i++ {
+		pkts = append(pkts, p.pktQ[i])
+	}
+	p.pktQ = []*P4RTPacketInfo{}
+
+	return p.pktCounters.RxPktCntr, pkts, nil
```

Comment:
There should be an indication that a timeout has occurred.
In case we are waiting for 10 packets, and we timeout in the middle, this routine would return less than 10 packets without any indication that a timeout has occurred. Perhaps the equivalent of ETIMEDOUT?
Also, a comment on top of this function to indicate that a ETIMEDOUT may contain packets is also recommended.

Reply:
Done

Reply:
Using `os.ErrDeadlineExceeded` which is a Timeout Error.